File: modules/reservation/src/main/java/com/media/bus/reservation/contract/dto/response/ContractResponse.java
package com.media.bus.reservation.contract.dto.response;

import com.media.bus.reservation.contract.entity.Contract;
import io.swagger.v3.oas.annotations.media.Schema;

import java.time.OffsetDateTime;
import java.util.UUID;

/// 계약 생성 응답 DTO.
@Schema(description = "계약 응답")
public record ContractResponse(

    @Schema(description = "계약 ID")
    UUID id,

    @Schema(description = "정류소 ID")
    UUID stopId,

    @Schema(description = "회원 ID")
    UUID memberId,

    @Schema(description = "계약명")
    String contractName,

    @Schema(description = "계약 상태")
    String status,

    @Schema(description = "계약 시작일")
    OffsetDateTime contractStartDate,

    @Schema(description = "계약 종료일")
    OffsetDateTime contractEndDate,

    @Schema(description = "생성일시")
    OffsetDateTime createdAt
) {

    /// Contract 엔티티로부터 응답 DTO를 생성하는 정적 팩토리 메서드.
    public static ContractResponse from(Contract contract) {
        return new ContractResponse(
            contract.getId(),
            contract.getStopId(),
            contract.getMemberId(),
            contract.getContractName(),
            contract.getStatus(),
            contract.getContractStartDate(),
            contract.getContractEndDate(),
            contract.getCreatedAt()
        );
    }
}

File: modules/reservation/src/main/kotlin/com/media/bus/reservation/contract/dto/response/ContractResponse.kt
package com.media.bus.reservation.contract.dto.response

import com.media.bus.reservation.contract.entity.ContractEntity
import com.media.bus.reservation.contract.entity.enums.ContractStatus
import io.swagger.v3.oas.annotations.media.Schema
import java.time.OffsetDateTime
import java.util.*

/** ## 계약 생성 응답 DTO */
@Schema(description = "계약 응답")
data class ContractResponse(
    @param:Schema(description = "계약 ID") val id: UUID,
    @param:Schema(description = "정류소 PK (UUID)") val stopId: UUID,
    @param:Schema(description = "정류소 번호 (STOPS_NO) — stop 서비스 장애 시 null", nullable = true)
    val stopNumber: String?,
    @param:Schema(description = "정류소 이름 — stop 서비스 장애 또는 삭제된 정류소일 경우 null", nullable = true)
    val stopName: String?,
    @param:Schema(description = "회원 ID (비회원 계약 시 null)", nullable = true) val memberId: UUID?,
    @param:Schema(description = "계약명") val contractName: String,
    @param:Schema(description = "계약 상태") val status: ContractStatus,
    @param:Schema(description = "계약 시작일") val contractStartDate: OffsetDateTime,
    @param:Schema(description = "계약 종료일") val contractEndDate: OffsetDateTime,
    @param:Schema(description = "생성일시") val createdAt: OffsetDateTime,
) {
    companion object {
        /**
         * ContractEntity로부터 응답 DTO를 생성한다.
         *
         * @param stopNumber stop 서비스에서 조회한 정류소 번호 (없으면 null)
         * @param stopName   stop 서비스에서 조회한 정류소 이름 (없으면 null)
         */
        fun from(
            contract: ContractEntity,
            stopNumber: String? = null,
            stopName: String? = null,
        ): ContractResponse = ContractResponse(
            id = contract.id.value,
            stopId = contract.stopId,
            stopNumber = stopNumber,
            stopName = stopName,
            memberId = contract.memberId,
            contractName = contract.contractName,
            status = contract.status,
            contractStartDate = contract.contractStartDate,
            contractEndDate = contract.contractEndDate,
            createdAt = contract.createdAt,
        )
    }
}
